lib/coherence/controllers/password_controller.ex

```diff
@@ -142,4 +122,37 @@ defmodule Coherence.PasswordController do
     |> Map.put("reset_password_token", nil)
     |> Map.put("reset_password_sent_at", nil)
   end
+
+  defp recover_password(conn, user_schema, nil, params) do
+    if Config.allow_silent_password_recovery_for_unknown_user do
+      if Config.mailer?() do
+        put_flash(conn, :info, Messages.backend().reset_email_sent())
+      else
+        put_flash(conn, :error, Messages.backend().mailer_required())
+      end
+      |> redirect_to(:password_create, params)
+    else
+      changeset = Controller.changeset :password, user_schema, user_schema.__struct__
+
+      conn
+      |> put_flash(:error, Messages.backend().could_not_find_that_email_address())
+      |> render("new.html", changeset: changeset)
+    end
+  end
+  defp recover_password(conn, user_schema, user, params) do
+    token = random_string 48
+    url = router_helpers().password_url(conn, :edit, token)
+    dt = Ecto.DateTime.utc
+
+    Config.repo.update! Controller.changeset(:password, user_schema, user,
+      %{reset_password_token: token, reset_password_sent_at: dt})
+
+    if Config.mailer?() do
+      send_user_email :password, user, url
+      put_flash(conn, :info, Messages.backend().reset_email_sent())
+    else
+      put_flash(conn, :error, Messages.backend().mailer_required())
+    end
```

Comment:
Would it make sense to extract this block to a separate private method since it's almost identical in both definitions of `recover_password/4`?

Reply:
I think this calls for a project-wide refactor since this pattern is for confirmation, invitation and unlock emails.

Reply:
This might be a good idea. However, there are two versions of the controller files. One lives in the coherence package space and the other is a generated file. So, if we create another module, we will probably need add a new generated file for the controllers generator.

Reply:
Yes, lets pull this block out into a separate defp. That will remove the nested if statement, which gave me some heartburn. @remiprev, I also think that we should document the new option in the readme and in one of the docmodule blocks to bring attention to this great new feature. When you have updates the PR with these two items, I'll merge.

Thanks for your contribution!!